### ISSUE TITLE
remove secrets (passwords etc) from settings

### ISSUE THREAD

**Issue by robkooper** [PecanProject/pecan#1379]
When writing the settings file it will save any passwords, the passwords should be masked when writing the settings file.

## Description
When writing the settings back to disk, the passwords (and potentially usernames) should be replaced with *****. This will prevent the settings file from being reloaded since the passwords are no longer in the settings file.

This will break how workflow.R is currently operating since it checks to see if the status has executed a stage. If not, it will run that stage, if it did run it will reload  the settings that was previously saved.

## Checks needed
Do the following stages actually modify the settings.xml
- [ ] TRAIT - get trait data from BETY
- [ ] CONFIG - write out config files for model runs
- [ ] ADVANCED - breaks the workflow to ask for user input (**NEEDS FIXING**)
- [ ] Benchmarking - loads additional settings files, any passwords here @bcow 
- [ ] multi site - loads additional settings files, any passwords here @rykelly

**Comment by mdietze** [comment]
The multisite (@rykelly) doesn't load additional settings files, it expands the current settings from the name of a sitegroup to all (or a random subset) of the sites in the group. Because it has the option to do a random subset, it's not exactly reproducible it it doesn't save the settings. However, it doesn't involve any passwords since it's not bringing in any new information

TRAIT + META-ANALYSIS together should modify settings because after the meta-analysis there's a new posterior distribution pointed to.

CONFIG should read from settings, but I'm not aware of it modifying anything

ADVANCED allows the modification of model-specific config files, but it doesn't change the pecan.xml. I'm not aware of it modifying the internal settings.

Benchmarking definitely modifies the settings, but doesn't require passwords.

In general, I think the only stage that brings in NEW information will be do.conversions, so the need for external passwords will probably be concentrated there in the short-term.  

Longer term, as we generalize the ingest of new data for visualization, PDA, SDA, and setting up new benchmarks there's definitely the potential for needing passwords. But I also see the set-up of all of these increasingly relying on interactive systems (e.g. Shiny).

**Comment by rykelly** [comment]
> The multisite (@rykelly) doesn't load additional settings files, it expands the current settings from the name of a sitegroup to all (or a random subset) of the sites in the group. Because it has the option to do a random subset, it's not exactly reproducible it it doesn't save the settings. However, it doesn't involve any passwords since it's not bringing in any new information

Yup. That's a good point about reproducibility though... we should give that function a random seed argument.

**Comment by ashiklom** [comment]
One option I'd like tho throw out is to leverage the capabilities of [Ansible](https://www.ansible.com/) for remote execution. The website looks flashy and corporate (because it's owned by Red Hat), but the underlying functionality of the command-line-based platform is open source. The idea is that you set up ["playbooks"](http://docs.ansible.com/ansible/playbooks_intro.html), which are simple YAML files describing command and the hosts on which to execute them.

I've played around a bit with this, and it seems like it solves the problem of SSH tunneling really effectively. You can specify hosts with a local file that looks something like this:

```
# hosts
[local]
localhost

[local:vars]
ansible_connection=local

[testpecan]
test-pecan.bu.edu ansible_connection=ssh ansible_user=ashiklom 

[geo]
geo.bu.edu ansible_connection=ssh ansible_user=ashiklom ansible_ssh_pass="MYSECRETPASSWORD"
```

...and then you can execute commands on a specific host by running:

```
ansible -i hosts local -m shell -a 'echo "Hello world"'
# localhost | SUCCESS | rc=0 >>
# Hello world

### Automatic SSH-key authentication for test-pecan
ansible -i hosts testpecan -m shell -a 'echo "Hello world"'
# test-pecan.bu.edu | SUCCESS | rc=0 >>
# Hello world

### Pass in SSH password from the "hosts" file
ansible -i hosts geo -m shell -a 'echo "Hello world"'
# geo.bu.edu | SUCCESS | rc=0 >>
# Hello world
```

As an extra layer of security, Ansible has [Vaults](http://docs.ansible.com/ansible/playbooks_vault.html) which are password-protected files for storing variables. So we could have users store their SSH passwords (e.g. `"MYSECRETPASSWORD"` from above) as variables in a password-protected vault file, and have each user create a local `.gitignore`d file containing the vault password (or pass that password in from the web interface each time).
